woocommerce-services.php

```diff
@@ -645,13 +645,40 @@ public function attach_hooks() {
 			add_action( 'admin_init', array( $this, 'load_admin_dependencies' ) );
 			add_filter( 'wc_connect_shipping_service_settings', array( $this, 'shipping_service_settings' ), 10, 3 );
 			add_action( 'woocommerce_email_after_order_table', array( $this, 'add_tracking_info_to_emails' ), 10, 3 );
+			add_action( 'save_post', array( $this, 'maybe_trigger_product_notice' ), 10, 2 );
 
 			$tracks = $this->get_tracks();
 			$tracks->init();
 
 			$this->taxjar->init();
 		}
 
+		public function maybe_trigger_product_notice( $post_id, $post ) {
+			if (
+				'publish' !== $post->post_status ||
+				'product' !== $post->post_type ||
+				! ( $product = wc_get_product( $post ) )
+			) {
+				return;
+			}
+
+			if ( $product->get_weight() && $product->get_length() && $product->get_width() && $product->get_height() ) {
+				// If details are all present, check all other products to see if notice should be disabled
+				$all_products_ready = true;
+				foreach ( wc_get_products( array( 'limit' => -1 ) ) as $product ) {
```

Comment:
This feels like **a lot** of work to do on product save. `limit => -1` always raises a flag for me in `WP_Query`s. _(This should be limited to only non-virtual products, for instance)_

I think a store-wide look at product shipping readiness makes more sense as part of our Status page.